conformance/utils/kubernetes/portset.go

```diff
@@ -0,0 +1,34 @@
+/*
+Copyright 2022 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package kubernetes
+
+import (
+	"sigs.k8s.io/gateway-api/apis/v1beta1"
+)
+
+// PortSet is a wrapper for handling sets of ports.
+type PortSet []v1beta1.PortNumber
```

Comment:
We have `Set` types available in Kubernetes utilities `k8s.io/apimachinery/pkg/utils/sets`, which you can see we already use in [conformance/utils/suite/suite.go:77](https://github.com/kubernetes-sigs/gateway-api/blob/2ae5227364ccf25652fb233554861ffd4c0601c8/conformance/utils/suite/suite.go#L77). Could we perhaps just use these?

Reply:
Actually, this changes the semantics to an unordered set, which makes the output non deterministic. Not necessarily a problem in practice, but it does make the tests significantly harder to write.

Reply:
Right, I consider this resolved at your discretion/preference.